core/ramatak/monetization_settings.cpp

```diff
@@ -33,7 +33,20 @@ void MonetizationSettings::delete_ad_unit(StringName p_ad_unit) {
 }
 
 Array MonetizationSettings::get_ad_units() const {
-	return ad_units;
+	String ad_unit_setting = "ramatak/monetization/ad_units";
+	ProjectSettings *project_settings = ProjectSettings::get_singleton();
+
+	Dictionary dict = project_settings->get(ad_unit_setting);
+	return dict.keys();
+}
+
+AdServer::AdType MonetizationSettings::get_ad_type(StringName p_ad_unit) const {
+	String ad_unit_setting = "ramatak/monetization/ad_units";
```

Comment:
This could just be a constant-equivalent I guess, given it's used across two different functions.

Reply:
Actually, I wonder if the above means there should be a helper function that gets the ad_unit_setting from the project_settings and just returns it as a dictionary, so you can replace the below with that helper function, and the above with helper_function.keys().

Reply:
It's just a few lines so I don't know about a helper function, but I definitely think an error-prone string constant should be moved to a constant.